src/types.d.ts

```diff
@@ -58,24 +78,300 @@ export interface AutoEncryptionOptions {
+  /** Corresponds to the write concern w Option. The w option requests acknowledgement that the write operation has propagated to a specified number of mongod instances or to mongod instances with specified tags. */
+  w?: number | 'majority';
+  /** Corresponds to the write concern wtimeout. wtimeoutMS specifies a time limit, in milliseconds, for the write concern. */
+  wtimeoutMS?: number;
+  /** Corresponds to the write concern j Option option. The journal option requests acknowledgement from MongoDB that the write operation has been written to the journal. */
+  journal?: boolean;
+  /** The level of isolation */
+  readConcernLevel?: ReadConcernLevels;
+  /** Specifies the read preferences for this connection */
+  readPreference?: ReadPreferenceModes;
+  /** Specifies, in seconds, how stale a secondary can be before the client stops using it for read operations. */
+  maxStalenessSeconds?: number;
+  /** Specifies the tags document as a comma-separated list of colon-separated key-value pairs.  */
+  readPreferenceTags?: string | string[] | { [key: string]: string };
+  /** Specify the database name associated with the user’s credentials. */
+  authSource?: string;
+  /** Specify the authentication mechanism that MongoDB will use to authenticate the connection. */
+  authMechanism?: AuthMechanisms;
+  /** Specify properties for the specified authMechanism as a comma-separated list of colon-separated key-value pairs. */
+  authMechanismProperties?: AuthMechanismProperties;
+  /** Set the Kerberos service name when connecting to Kerberized MongoDB instances. This value must match the service name set on MongoDB instances to which you are connecting. */
+  gssapiServiceName?: string;
+  /** The size (in milliseconds) of the latency window for selecting among multiple suitable MongoDB instances. */
+  localThresholdMS?: number;
+  /** Specifies how long (in milliseconds) to block for server selection before throwing an exception.  */
+  serverSelectionTimeoutMS?: number;
+  /** When true, instructs the driver to scan the MongoDB deployment exactly once after server selection fails and then either select a server or raise an error. When false, the driver blocks and searches for a server up to the serverSelectionTimeoutMS value. */
+  serverSelectionTryOnce?: boolean;
+  /** heartbeatFrequencyMS controls when the driver checks the state of the MongoDB deployment. Specify the interval (in milliseconds) between checks, counted from the end of the previous check until the beginning of the next one. */
+  heartbeatFrequencyMS?: number;
+  /** Specify a custom app name. */
+  appName?: string;
+  /** Enables retryable reads. */
+  retryReads?: boolean;
+  /** Enable retryable writes. */
+  retryWrites?: boolean;
+  /** Allow a driver to force a Single topology type with a connection string containing one host */
+  directConnection?: boolean;
+}
+
+export interface MongoClientOptions extends MongoURIOptions, BSONSerializeOptions {
```

Comment:
lookin really good, but I think based on our conversation yesterday we want to put this definition into `mongo_client.ts`. The `types.d.ts` file is currently where "global" or "missing" types are going (auto encrypter is not typed for instance)